### ISSUE TITLE
Azure Service Fabric host

### ISSUE THREAD

**Issue by ppittle** [Particular/NServiceBus#3759]
Can you add support for running NServiceBus on Azure Service Fabric (inside Azure Service Fabric Services)?

I've done some research into running NSB on ASF and it's possible, but requires writing some plumbing code:
- I was able to create an ASF `ICommunicationListener` that created an `EndpointInstance` and tied the endpoint lifecycle to the communication listener (via it's Open, Close, Abort methods).

And then optionally, it would be nice to have a SagaPersistor that could plug into's ASF's `IReliableStateManager` for fast sage persistance.

**Comment by andreasohlund** [comment]
Sounds interesting, can you share the code?

**Comment by danielmarbach** [comment]
@ppittle Would that help?

https://github.com/ParticularLabs/NServiceBus.ServiceFabric

It is in labs and not officially supported yet.

**Comment by yvesgoeleven** [comment]
I got a similar question via twitter https://twitter.com/VeljkoKokanovic/status/730757448055857152

**Comment by yvesgoeleven** [comment]
Got a subtle hint on the Labs repo as well https://github.com/ParticularLabs/NServiceBus.ServiceFabric/issues/1#issuecomment-221038137

**Comment by ppittle** [comment]
@danielmarbach - The NServiceBus.ServiceFabric project looks like a good start, but it doesn't have a "host". Should I re-post my comment on that repo?

**Comment by danielmarbach** [comment]
@ppittle No it is OK to leave it here. we'll keep track of it for you

**Comment by yvesgoeleven** [comment]
@ppittle What would you expect the host to do?

**Comment by ppittle** [comment]
Here's most of the plumbing I needed to wire up to get NSB running on ASF:

https://gist.github.com/ppittle/37c5a37429a802f90e2b5fd5bca93410

So this Starts/Stops an Endpoint when ASF Starts/Stops the service.

**Comment by danielmarbach** [comment]
@ppittle We are working on a Service Fabric persistence for Outbox and Sagas. We will not address Timeouts, Subscriptions and Gateway deduplication. Timeouts and Subscriptions become more and more part of the transport. For gateway deduplication mostly the InMemory one is good enough. 

The work is ongoing and can be observed on this repo https://github.com/Particular/NServiceBus.Persistence.ServiceFabric

Before we talked the persistence we addressed the biggest topic for SF which is partition affine routing. There is already a fully shaped sample based on alpha release of Core 6.2 available on our docs site.

https://docs.particular.net/samples/azure/azure-service-fabric-routing/

When the persistence is done we will write guidance and best practices for hosting NSB endpoint in SF. Stay tuned
